src/irmin-pack/unix/append_only_file.ml

```diff
@@ -164,6 +166,6 @@ module Make (Io : Io.S) = struct
         assert (Buffer.length rw_perm.buf < rw_perm.auto_flush_threshold);
         Buffer.add_string rw_perm.buf s;
         if Buffer.length rw_perm.buf >= rw_perm.auto_flush_threshold then (
-          rw_perm.auto_flush_callback t;
+          flush t |> rw_perm.auto_flush_callback;
           assert (empty_buffer t))
```

Comment:
I am on the fence about leaving this assert in if we are taking care of flush internally now. Open to opinions.